test/clipping.jl

```diff
@@ -0,0 +1,151 @@
+
+
+@testset "Clipping" begin
+  @testset "SutherlandHodgman" begin
+    # triangle
+    poly = Triangle(P2(6, 2), P2(3, 5), P2(0, 2))
+    other = Quadrangle(P2(5, 0), P2(5, 4), P2(0, 4), P2(0, 0))
+
+    clipped = clip(poly, other, SutherlandHodgman())
+    
+    @test all(vertices(clipped) .≈ [
+      P2(5, 3),
+      P2(4, 4),
+      P2(2, 4),
+      P2(0, 2),
+      P2(5, 2)
+    ])
+
+    # octagon
+    poly = Octagon(
+      P2(8,-2),
+      P2(8, 5),
+      P2(2, 5),
+      P2(4, 3),
+      P2(6, 3),
+      P2(4, 1),
+      P2(2, 1),
+      P2(2,-2)
+    )
+
+    other = Quadrangle(P2(5, 0), P2(5, 4), P2(0, 4), P2(0, 0))
+    clipped = clip(poly, other, SutherlandHodgman())
+    @test all(vertices(clipped) .≈ [
+      P2(3, 4),
+      P2(4, 3),
+      P2(5, 3),
+      P2(5, 2),
+      P2(4, 1),
+      P2(2, 1),
+      P2(2, 0),
+      P2(5, 0),
+      P2(5, 4)
+    ])
+
+    # "random"
+    poly = Ngon(P2(0, 0), rand(P2, 10)...)
```

Comment:
```suggestion
    poly = Ngon(P2(0, 0), rand(rng, P2, 10)...)
```

Reply:
Random tests are tricky in general, and hard to debug later. Can you replace this one with another case that is more predictable?